ls1intum/Artemis
# `Communication`: Create default channels when course is created

### Checklist
#### General


#### Server

### Motivation and Context



### Description

This is the first step in our path to merge the communication tab with the messages tag. With this PR we create the 4 default channels, when a course is created:
- `announcement`
- `tech-support`
- `random`
- `organization`

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students

1. Log in to Artemis as an admin
2. Navigate to Course Administration
3. Create a new course
4. Add a user to the course
5. Login as this user
6. Go to the course messages tab, browse channels and see, if there are 4 new channels available: `announcement`, `tech-support`, `random`, `organization`
7. Only the `announcement` should be a "announcement" channel
8. All channels should be public
9. Join the channels
10. Add other users (1 student, 1 tutor, 1 additional instructor) to the course
11. Check if the other users are automatically added to all 4 default channels
12. Create another course with the same student/tutor/instructor groups
13. Check if all users are automatically added to the 4 default channels of the new course

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage





#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [DefaultChannelType.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4786-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.enumeration/DefaultChannelType.html) | 100% | ✅ |
| [CourseRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4786-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository/CourseRepository.html) | 81% | ✅ |
| [ChannelService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4786-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis.conversation/ChannelService.html) | 96% | ✅ |
| [CourseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4786-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/CourseResource.html) | 94% | ✅ |
| [AdminCourseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4786-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.admin/AdminCourseResource.html) | 100% | ✅ |
| [ChannelResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4786-28/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.metis.conversation/ChannelResource.html) | 91% | ✅ |



### Screenshots
